test/token/ERC777/behaviors/ERC777Mintable.behavior.js

```diff
@@ -0,0 +1,56 @@
+const { BN, constants, expectEvent, expectRevert } = require('@openzeppelin/test-helpers');
+const { ZERO_ADDRESS } = constants;
+
+const { expect } = require('chai');
+
+function shouldBehaveLikeERC777Mintable (minter, data, operatorData, [other]) {
+  describe('as a mintable token', function () {
+    describe('mintInternal', function () {
+      const amount = new BN(100);
+
+      context('when the sender has minting permission', function () {
+        const from = minter;
+
+        context('for a zero amount', function () {
+          shouldMint(new BN(0));
+        });
+
+        context('for a non-zero amount', function () {
+          shouldMint(amount);
+        });
+
+        function shouldMint (amount) {
+          beforeEach(async function () {
+            ({ logs: this.logs } = await this.token.mint(other, other, amount, data, operatorData, { from }));
+          });
+
+          it('mints the requested amount', async function () {
+            expect(await this.token.balanceOf(other)).to.be.bignumber.equal(amount);
+          });
+
+          it('emits a mint and a transfer event', async function () {
+            expectEvent.inLogs(this.logs, 'Transfer', {
```

Comment:
We should check here the [`Minted`](https://docs.openzeppelin.com/contracts/2.x/api/token/erc777#IERC777-Minted-address-address-uint256-bytes-bytes-) event is emitted with the correct `data`, `operatorData`, etc. 

I wouldn't go into more detail than that (such as checking the receive hooks are called).